### DISCUSSION TITLE
Alchemy limit reached/goes down

### DISCUSSION THREAD

**Discussion by missphyrgames** [wevm/wagmi#1017] [Q&A]
I was wondering what would happen if the alchemy api key I provide went over the limit that they provide. I've noticed the wagmi library makes a lot of calls to the alchemy endpoint and so I'm worried it will hit the capacity. I've also added an Infura api key too but what happens if they both hit the capacity?

I'm creating a minting dapp and so it's likely I will hit the limit (At least I think it will - multiple contract reads with watch set to true). Is it also possible to use the provider given by Metamask using the wagmi library? Thank you! (I'm still learning everything so apologies for the unstructured query)

**Comment by thecil** [comment]
Same here, would like to know if there is a way to config wagmi for rate limit on alchemy or infura, any suggestion will be highly appreciated
